app/javascript/components/Game.jsx

```diff
@@ -111,8 +111,14 @@ export const Game = () => {
 
             </div>}
 
-            {data?.sessions ? <Sessions data={user?.sessions?.filter(s => s.game_id == data?.id).concat(user?.shared_sessions?.filter(s => s.game_id == data?.id))} 
-                user={user} game_id={data?.id} setter={setNewSession} /> : ''}
+            {data?.sessions ? <Sessions data={
+                data?.sessions.filter(session => session.user_id == user.id ||
+                    session.session_shares?.map(share => share.collaborator_id)
+                        .includes(user.id))
+                }
```

Comment:
I know that you mentioned potentially doing more refactoring here. Just wanted to say that pulling this filter method out into a const defined higher up in the component may not change any of what is happening, but it allows you to name it.

Naming things is a powerful way to make code easier to read. 

```suggestion
const sessionUserOwnsOrCollaboratedIn = data?.sessions.filter(session => 
  session.user_id == user.id ||
  session.session_shares?.map(share => share.collaborator_id).includes(user.id))
            
...
{data?.sessions && <Sessions data={sessionUserOwnsOrCollaboratedIn} user={user} game_id={data?.id} setter={setNewSession} />}

```

That suggestion also changes the render logic just a bit. Instead of doing 

```
{somedata.exists ? <MyComponent /> : ""}
```

You can use 

```
{somedata.exists && <MyComponent />}
```

When you use `&&` it will only render the component if the statement before the `&&` is truthy.